src/js/jsx/tools/GuidesOverlay.jsx

```diff
@@ -77,7 +78,7 @@ define(function (require, exports, module) {
         },
 
         shouldComponentUpdate: function (nextProps, nextState) {
-            return !Immutable.is(this.state.uiState, nextState.uiState) ||
+            return !_.isEqual(this.state.uiState, nextState.uiState) ||
```

Comment:
This was an unrelated bug; `uiState` is not an immutable object. At some point we should convert it to be an immutable object, though!